src/compiler/transformers/classFields.ts

```diff
@@ -778,7 +778,7 @@ namespace ts {
             }
 
             const initializer = property.initializer || emitAssignment ? visitNode(property.initializer, visitor, isExpression)
-                : hasModifier(getOriginalNode(property), ModifierFlags.ParameterPropertyModifier) && isIdentifier(propertyName) ? propertyName
+                : hasModifier(getOriginalNode(property), ModifierFlags.ParameterPropertyModifier) && isIdentifier(propertyName) && isParameter(getOriginalNode(property)) ? propertyName

```

Comment:
suggestion: you can replace the whole condition with `isParameterPropertyDeclaration(getOriginalNode(property))`

Reply:
I'm not that familiar with the TS codebase but that sounds like it would be equivalent to `hasModifier(getOriginalNode(property), ModifierFlags.ParameterPropertyModifier)`?

Reply:
It also checks that the parent is a constructor declaration.